README.md
# pokdeng-project_game-python

![GitHub Logo](https://shields.io/badge/Python-3-blue)

โปรเจคนี้เขียนขึ้นเพื่อเป็นการฝึกการเขียนโปรแกรมภาษา Python

# Updatelog
![GitHub Logo](https://shields.io/badge/Version-1.0-green)
 - ก๊อปปี้ข้อมูลเด็คมาจากต้นแบบ เพื่อไม่ให้กระทบกับเด็คหลักเมื่อผู้เล่นสุ่มหยิบไพ่
 - สุ่มไพ่ให้ผู้เล่น 2 ใบ และ บอท 2 ใบ สุ่มจากเด็คที่สร้างใหม่ และไม่สามารถได้ใบซ้ำกัน
 - เช็คแต้มของกาด, เอาไพ่ทั้งสองบวกกันและหาร 10 โดย 10 J Q K มีแต้มไพ่ท่ากับ 0
 - เช็คแต้มของผู้เล่น และบอท ว่าใครแต้มเยอะกว่าและเป็นผู้ชนะ หรือแต้มเสมอกัน และหากแต้มมากกว่าหรือเท่ากับ 8 จะถือว่าเป็นป็อก
 
 ![GitHub Logo](https://shields.io/badge/Version-1.1-green)
 - เพิ่มการสุ่มไพ่ใบที่ 3 ของผู้เล่นและบอท ถ้าหากไม่มีใครป็อกจะให้เลือกว่าจะสุ่มไพ่เพิ่มหรือไม่สุ่ม
 - เช็คป็อกเด้ง
 - แต้ม 8-9 จากไพ่ 3 ใบไม่ถือว่าเป็นป็อก ( 9 จากรวมแต้ม 3 ใบ แพ้ 9 จากรวมแต้ม 2 ใบ )
 
 ![GitHub Logo](https://shields.io/badge/Version-1.2-green)
- เช็คเด้ง(2-3), เรียงฟลัช, เรียง, สามเหลือง, ตอง

![GitHub Logo](https://shields.io/badge/Version-2.5-green)<br>
- เพิ่มหน้าต่าง Ui
- ปรับการเก็บข้อมูลการ์ดผู้เล่นและบอท
- เปลี่ยนการจั๋วการ์ดเพิ่มหรือเริ่มเกมใหม่ ไม่ต้องพิมพ์ส่งค่าอินพุต กดปุ่มที่หน้าต่างเกมได้เลย

# Todo
![GitHub Logo](https://shields.io/badge/NextUpdate->>>-orange)
<!-- - ปิดการใช้คอลโทรน -->
<!-- - แสดงข้อความกดปุ่มในหน้าต่างเกม -->
<!-- - เพิ่มข้อความแสดงสกอบอร์ดในหน้าต่างเกม -->
- เพิ่มเก้าอี้ผู้เล่นเป็น 5 เก้าอี้
- ผู้เล่นสามารถเลือกนั่งได้
- สามารถเพิ่มบอทในเก้าอี้ได้
- ทำระบบออนไลน์

# Rule Game
ประเภทไพ่ในมือ | จำนวนไพ่ | แต้ม | เด้ง | ไพ่ในมือ | แต้ม | เด้ง | ไพ่ในมือ | แต้ม | เด้ง | หมายเหตุ
------|------|------|------|------|------|------|------|------|------|------
ป๊อก | 2 | 8-9 | 1-2 | J♦ 9♥ | 9 | 1 | 9♠ 9♦ | 8 | 2
ตอง | 3 | - | 5 | 7♠ 7♥ 7♣ | 1 | 5 | K♣ K♦ K♥ | 0 | 5
เรียงฟลัช | 3 | - | 5 | 5♦ 6♦ 7♦ | 8 | 5 | 2♥ 3♥ 4♥ | 9 | 5 | Q♠ K♠ A♠<br>ไม่นับเป็นเรียง
เรียง | 3 | - | 3 | 2♠ 3♦ 4♣ | 9 | 3 | 7♣ 8♣ 9♥ | 4 | 3 | Q♣ K♠ A♥<br>ไม่นับเป็นเรียง
สามเหลือง | 3 | 0 | 3 | Q♣ J♣ J♦ | 0 | 3 | J♠ J♦ K♠ | 0 | 3 | K♣ K♠ A♥<br>ไม่นับเป็นสามเหลือง
ปกติ | 2-3 | 0-9 | 1-3 | A♣ 4♣ 5♣ | 9 | 3 | 5♥ 7♥ 7♠ | 9 | 1

ChangeLog.md
# ChangeLog

# Version 1
![GitHub Logo](https://shields.io/badge/Version-1.0-green)
 - ก๊อปปี้ข้อมูลเด็คมาจากต้นแบบ เพื่อไม่ให้กระทบกับเด็คหลักเมื่อผู้เล่นสุ่มหยิบไพ่
 - สุ่มไพ่ให้ผู้เล่น 2 ใบ และ บอท 2 ใบ สุ่มจากเด็คที่สร้างใหม่ และไม่สามารถได้ใบซ้ำกัน
 - เช็คแต้มของกาด, เอาไพ่ทั้งสองบวกกันและหาร 10 โดย 10 J Q K มีแต้มไพ่ท่ากับ 0
 - เช็คแต้มของผู้เล่น และบอท ว่าใครแต้มเยอะกว่าและเป็นผู้ชนะ หรือแต้มเสมอกัน และหากแต้มมากกว่าหรือเท่ากับ 8 จะถือว่าเป็นป็อก
 
![GitHub Logo](https://shields.io/badge/Version-1.1-green)
 - เพิ่มการสุ่มไพ่ใบที่ 3 ของผู้เล่นและบอท ถ้าหากไม่มีใครป็อกจะให้เลือกว่าจะสุ่มไพ่เพิ่มหรือไม่สุ่ม
 - เช็คป็อกเด้ง
 - แต้ม 8-9 จากไพ่ 3 ใบไม่ถือว่าเป็นป็อก ( 9 จากรวมแต้ม 3 ใบ แพ้ 9 จากรวมแต้ม 2 ใบ )
 
![GitHub Logo](https://shields.io/badge/Version-1.2-green)
 - เช็คเด้ง(2-3), เรียงฟลัช, เรียง, สามเหลือง, ตอง
 
![GitHub Logo](https://shields.io/badge/Version-1.3-green)
 - ไพ่ 9 2 ใบเป็น 8 2 เด้ง (3 เด้งเหมือนกัน)

# Version 2
![GitHub Logo](https://shields.io/badge/Version-2.1-green)<br>
[Ui without System]
 - ขึ้นรูปร่างหน้า ui

![GitHub Logo](https://shields.io/badge/Version-2.2-green)<br>
[Ui without System]
 - แก้ไข้อาการหน้าต่างค้าง

![GitHub Logo](https://shields.io/badge/Version-2.4-green)<br>
- เพิ่มระบบการเล่น
- ปรับการเก็บข้อมูลการ์ดผู้เล่นและบอท

![GitHub Logo](https://shields.io/badge/Version-2.5-green)<br>
- เปลี่ยนการจั๋วการ์ดเพิ่มหรือเริ่มเกมใหม่ ไม่ต้องพิมพ์ส่งค่าอินพุต กดปุ่มที่หน้าต่างเกมได้เลย

![GitHub Logo](https://shields.io/badge/Version-2.6-green)<br>
- แสดงสกอบอร์ดในหน้าต่างเกม
- แสดงข้อความกดปุ่มในหน้าต่างเกม
- ขึ้นข้อความว่าใครชนะ


test.py
xcard = ['23A', '234', '345', '456', '567', '678', '789', '1089', '109J', '10JQ']
scp = 'dd'
scb = 'dd'
for xc in range(len(xcard)):
    if xcard[xc] == scp or xcard[xc] == scb:
        xcheck = 1
        break

game.py
import random
from time import sleep
import os
from Modules.module_deck import *
from Modules.module_checkwin import *
import pygame

clear = lambda: os.system('cls')

display_status = ['None']
win_status = ['None']
status_table = 'hide'
deck_loop = []
cardloca = [
    {
        'name':'Player',
        'win':0,
        'cards':[
            NoneCard, NoneCard, NoneCard
        ]
    },
    {
        'name':'Dealer',
        'win':0,
        'cards':[
            NoneCard, NoneCard, NoneCard
        ]
    }
]

def main():
    clear()
    global deck_loop
    deck_loop = create_new_deck()
    random_card()

def random_card():
    global status_table
    status_table = 'hide'
    for x in range(2):
        for i in range(2):
            randomcard = random.randint(0, len(deck_loop)-1)
            # print('[SYS] {} Draw'.format(cardloca[i]['name']))
            cardloca[i]['cards'][x] = deck_loop[randomcard]
            deck_loop.pop(randomcard)
            sleep(1)
    sleep(1)
    check_win(2)
    
def player_third_card():
    check_press_key = True
    if point_player < 8:
        display_status[0] = 'Tcard'
        while check_press_key:
            pygame.time.delay(5)
            keys = pygame.key.get_pressed()
            if keys[pygame.K_d]:
                display_status[0] = 'None'
                check_press_key = False
                randomcard = random.randint(0, len(deck_loop)-1)
                cardloca[0]['cards'][2] = deck_loop[randomcard]
                deck_loop.pop(randomcard)
                print('[SYS] Player Draw more')
                sleep(1)
            if keys[pygame.K_s]:
                display_status[0] = 'None'
                check_press_key = False
                print('[SYS] Player Stay')
                sleep(1)
        bot_third_card()
    
def bot_third_card():
    if point_bot < 5:
        randomcard = random.randint(0, len(deck_loop)-1)
        print('[SYS] BOT Draw more')
        cardloca[1]['cards'][2] = deck_loop[randomcard]
        deck_loop.pop(randomcard)
        sleep(1)
        check_win(3)
    else:
        print('[SYS] BOT Stay')
        sleep(1)
        check_win(3)

def check_win(x):
    global point_player, point_bot, status_table
    point_player = (cardloca[0]['cards'][0]['Point'] + cardloca[0]['cards'][1]['Point'] + cardloca[0]['cards'][2]['Point']) % 10
    point_bot = (cardloca[1]['cards'][0]['Point'] + cardloca[1]['cards'][1]['Point'] + cardloca[1]['cards'][2]['Point']) % 10
    Mcheckwin = Module_check_win(x, cardloca[0]['cards'], point_player, cardloca[1]['cards'], point_bot)
    cardloca[0]['win'] = cardloca[0]['win'] + Mcheckwin[0]
    cardloca[1]['win'] = cardloca[1]['win'] + Mcheckwin[1]
    win_status[0] = Mcheckwin[3]
    if Mcheckwin[2] == 1:
        player_third_card()
    else:
        status_table = 'show'
        continue_game()

def continue_game():
    global point_player, point_bot, deck_loop
    check_press_key = True
    display_status[0] = 'Continue'
    while check_press_key:
        pygame.time.delay(5)
        keys = pygame.key.get_pressed()
        if keys[pygame.K_c]:
            win_status[0] = 'None'
            display_status[0] = 'None'
            check_press_key = False
            cardloca[0]['cards'] = [NoneCard, NoneCard, NoneCard]
            cardloca[1]['cards'] = [NoneCard, NoneCard, NoneCard]
            sleep(1.5)
            deck_loop = create_new_deck()
            random_card()

def status_show_cards():
    return status_table

def score_win():
    return cardloca[0]['win'], cardloca[1]['win']

def status_game():
    return display_status[0]

def status_win():
    return win_status[0]

main.py
# -*- coding: utf-8 -*-

import pygame
import pygame.freetype
from threading import Thread
from game import *
pygame.init()

GAME_FONT = pygame.freetype.Font("tahomabd.ttf", 24)

pygame.display.set_caption('Pokdeng Game Card')
icon = pygame.image.load('assets/icon.png')
pygame.display.set_icon(icon)
# # 720p
screen = pygame.display.set_mode((1280, 720))

bgload = pygame.image.load('assets/background.png')
bg = pygame.transform.scale(bgload, (1280, 720))
# # 720p

imgbackcard = 'assets/backcard.jpg'
load_backcard = pygame.image.load(imgbackcard)
position_dealer = [(528, 168), (548, 188), (573, 203)]
position_player1 = [(553, 423), (573, 443), (593, 463)]

card_dealer = ['None', 'None', 'None']
card_player1 = ['None', 'None', 'None']

def showcard(status):
    username = 'Player'
    if status == 'hide':
        # Player 1
        if username == cardloca[0]['name']:
            for i in range(3):
                load_card = pygame.image.load(cardloca[0]['cards'][i]['Imgage'])
                card_player1[i] = pygame.transform.scale(load_card, (90, 135))
        else:
            for i in range(3):
                if cardloca[0]['cards'][i]['Card'] != 'None':
                    card_player1[i] = pygame.transform.scale(load_backcard, (90, 135))
                else:
                    load_card = pygame.image.load(cardloca[0]['cards'][i]['Imgage'])
                    card_player1[i] = pygame.transform.scale(load_card, (90, 135))
        for i in range(3):
            if cardloca[1]['cards'][i]['Card'] != 'None':
                card_dealer[i] = pygame.transform.scale(load_backcard, (90, 135))
            else:
                load_card = pygame.image.load(cardloca[1]['cards'][i]['Imgage'])
                card_dealer[i] = pygame.transform.scale(load_card, (90, 135))

    elif status == 'show':
        for i in range(3):
            load_card = pygame.image.load(cardloca[0]['cards'][i]['Imgage'])
            card_player1[i] = pygame.transform.scale(load_card, (90, 135))
        for i in range(3):
            load_card = pygame.image.load(cardloca[1]['cards'][i]['Imgage'])
            card_dealer[i] = pygame.transform.scale(load_card, (90, 135))
    else:
        print('Show Card Error')

posX = 400
posY = 400

if __name__ == "__main__":
    Thread(target=main).start()
    run = True
    while run:
        x = score_win()
        dis_stu = status_game()
        win_stu = status_win()
        showcard(status_show_cards())
        pygame.time.delay(30)
        
        for eventget in pygame.event.get():
            pygame.display.flip()
            if eventget.type == pygame.QUIT:
                run = False
        ## ไว้สำหรับหาค่า X Y
        keys = pygame.key.get_pressed()
        if keys[pygame.K_LEFT]:
            posX-=5
        if keys[pygame.K_RIGHT]:
            posX+=5
        if keys[pygame.K_UP]:
            posY-=5
        if keys[pygame.K_DOWN]:
            posY+=5
        if keys[pygame.K_SPACE]:
            print('X {}\nY {}'.format(posX, posY))
        ## ไว้สำหรับหาค่า X Y

        screen.blit(bg, (0, 0))
        screen.blit(card_player1[0], position_player1[0])
        screen.blit(card_player1[1], position_player1[1])
        screen.blit(card_player1[2], position_player1[2])
        
        screen.blit(card_dealer[0], position_dealer[0])
        screen.blit(card_dealer[1], position_dealer[1])
        screen.blit(card_dealer[2], position_dealer[2])
        score_player, rect = GAME_FONT.render('Player Win:{}'.format(x[0]), (255, 255, 255))
        screen.blit(score_player, (330, 75))
        score_bot, rect = GAME_FONT.render('Bot Win:{}'.format(x[1]), (255, 255, 255))
        screen.blit(score_bot, (330, 120))
        # text_surface3, rect = GAME_FONT.render('เทสภาษาไทย', (255, 255, 255))
        # screen.blit(text_surface3, (posX, posY))
        if dis_stu == 'Tcard':
            text_surface3, rect = GAME_FONT.render('กดปุ่ม [D]จั๋วไพ่เพิ่ม [S]ไม่จั๋ว', (255, 255, 255))
            screen.blit(text_surface3, (515, 355))
        elif dis_stu == 'Continue':
            text_surface3, rect = GAME_FONT.render('กดปุ่ม [C] เริ่มเกมใหม่', (255, 255, 255))
            screen.blit(text_surface3, (520, 355))
        if win_stu != 'None':
            text_win, rect = GAME_FONT.render('{} Win'.format(win_stu), (255, 255, 255))
            screen.blit(text_win, (370, 210))
        text_surface, rect = GAME_FONT.render('เทสภาษาไทย อิอ ตี้ ทั่ว อี้  อี่', (0, 0, 0))
        screen.blit(text_surface, (posX, posY))
        pygame.display.update()
    pygame.quit()

logic1.py
player = [
    {'Card':'8'},
    {'Card':'9'},
    {'Card':'7'}
    ]

psort = []
for i in player:
    psort.append(i['Card'])
psort.sort()
scp = psort[0]+psort[1]+psort[2]

bsort = []
for x in bot:
    bsort.append(i['Card'])
bsort.sort()
scb = bsort[0]+bsort[1]+bsort[2]

if scp != scb:
    if scp == '23A' or scp == '234' or scp == '345' or scp == '456' or scp == '567' or scp == '678' or scp == '789' or scp == '1089' or scp == '109J' or scp == '10JQ':
        if player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
            print('เรียงฟลัช')
        else:
            print('เรียง')
    elif scb == '23A' or scb == '234' or scb == '345' or scb == '456' or scb == '567' or scb == '678' or scb == '789' or scb == '1089' or scb == '109J' or scb == '10JQ':
        if bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
            print('เรียงฟลัช')
        else:
            print('เรียง')
    elif scp == 'JKQ':
        print('สามเหลือง')
    elif scb == 'JKQ':
        print('สามเหลือง')
else:
    if player[0]['Suit'] == player[1]['Suit'] and player[2]['Card'] == 'None':
        print('สองเด้ง')
    elif player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
        print('สามเด้ง')
    else:
        print('ปกติ')

logic2.py
# player = [
#     {'Card':'8'},
#     {'Card':'9'},
#     {'Card':'7'}
#     ]

psort = []
for i in player:
    psort.append(i['Card'])
psort.sort()
scp = psort[0]+psort[1]+psort[2]
print(scp)

bsort = []
for x in bot:
    bsort.append(i['Card'])
bsort.sort()
scb = bsort[0]+bsort[1]+bsort[2]

xcard = ['23A', '234', '345', '456', '567', '678', '789', '1089', '109J', '10JQ']

if scp != scb:
    for i in range(len(xcard)):
        if scp == xcard[i]:
            if player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
                print('Pเรียงฟลัช')
            else:
                print('Pเรียง')
        elif scb == xcard[i]:
            if bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
                print('Bเรียงฟลัช')
            else:
                print('Bเรียง')
        elif scp == 'JKQ':
            print('Pสามเหลือง')
        elif scb == 'JKQ':
            print('Bสามเหลือง')
elif scp == scb:
    print('เสมอ')
else:
    if point_player > point_bot:
        if player[0]['Suit'] == player[1]['Suit'] and player[2]['Card'] == 'None':
            print('Pสองเด้ง')
        elif player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
            print('Pสามเด้ง')
        else:
            print('Pปกติ')
    elif point_player < point_bot: 
        if bot[0]['Suit'] == bot[1]['Suit'] and bot[2]['Card'] == 'None':
            print('Bสองเด้ง')
        elif bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
            print('Bสามเด้ง')
        else:
            print('Bปกติ')
    elif point_player == point_bot:
        print('เสมอ')

checkwin.py
from main import player, point_player, bot, point_bot

def check_win(x):
    if x == 2:
        CardPlayer = '\t' + player[0]['Emoji'] + '\t' + player[1]['Emoji']
        CardBot = '\t' + bot[0]['Emoji'] + '\t' + bot[1]['Emoji']
        if point_player >= 8 or point_bot >= 8:
            if point_player > point_bot:
                print('Your Point :' , point_player, '\nYour Card : ' , CardPlayer)
                print('Bot Point :' , point_bot, '\nBot Card : ' , CardBot)
                if player[0]['Suit'] == player[1]['Suit']:
                    print('You Win Pok{}Deng'.format(point_player))
                    player_win = player_win + 1
                else:
                    print('You Win Pok{}'.format(point_player))
                    player_win = player_win + 1
            elif point_player < point_bot: 
                print('Your Point :' , point_player, '\nYour Card : ' , CardPlayer)
                print('Bot Point :' , point_bot, '\nBot Card : ' , CardBot)
                if bot[0]['Suit'] == bot[1]['Suit']:
                    print('Bot Win Pok{}Deng'.format(point_bot))
                    bot_win = bot_win + 1
                else:
                    print('Bot Win Pok{}'.format(point_bot))
                    bot_win = bot_win + 1
            elif point_player == point_bot:
                print('Your Point :' , point_player, '\nYour Card : ' , CardPlayer)
                print('Bot Point :' , point_bot, '\nBot Card : ' , CardBot)
                print('Draw')
            player.clear()
            bot.clear()
            print('Clear Card on hand')
            sleep(2)
            continue_game()
        player_third_card()
    if x == 3:
        psort = []
        for i in player:
            psort.append(i['Card'])
        psort.sort()
        scp = psort[0]+psort[1]+psort[2]
        print(scp)

        bsort = []
        for x in bot:
            bsort.append(i['Card'])
        bsort.sort()
        scb = bsort[0]+bsort[1]+bsort[2]

        xcard = ['23A', '234', '345', '456', '567', '678', '789', '1089', '109J', '10JQ']

        if scp != scb:
            for i in range(len(xcard)):
                if scp == xcard[i]:
                    if player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
                        print('Pเรียงฟลัช')
                    else:
                        print('Pเรียง')
                elif scb == xcard[i]:
                    if bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
                        print('Bเรียงฟลัช')
                    else:
                        print('Bเรียง')
                elif scp == 'JKQ':
                    print('Pสามเหลือง')
                elif scb == 'JKQ':
                    print('Bสามเหลือง')
        elif scp == scb:
            print('เสมอ')
        else:
            if point_player > point_bot:
                if player[0]['Suit'] == player[1]['Suit'] and player[2]['Card'] == 'None':
                    print('Pสองเด้ง')
                elif player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
                    print('Pสามเด้ง')
                else:
                    print('Pปกติ')
            elif point_player < point_bot: 
                if bot[0]['Suit'] == bot[1]['Suit'] and bot[2]['Card'] == 'None':
                    print('Bสองเด้ง')
                elif bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
                    print('Bสามเด้ง')
                else:
                    print('Bปกติ')
            elif point_player == point_bot:
                print('เสมอ')

Modules/module_checkwin.py
def Module_check_win(x, player, point_player, bot, point_bot):
    player_third_card = 0
    player_win = 0; bot_win = 0
    win_stu = 'None'
    CardPlayer = '\t' + player[0]['Emoji'] + '\t' + player[1]['Emoji'] + '\t' + player[2]['Emoji']
    CardBot = '\t' + bot[0]['Emoji'] + '\t' + bot[1]['Emoji'] + '\t' + bot[2]['Emoji']
    if x == 2:
        if point_player >= 8 or point_bot >= 8:
            if point_player > point_bot:
                print('Your Point :\t' , point_player, '\nYour Card : ' , CardPlayer)
                print('Bot Point :\t' , point_bot, '\nBot Card : ' , CardBot)
                if player[0]['Suit'] == player[1]['Suit'] or player[0]['Card'] == player[1]['Card']:
                    print('Player win Pok{} (2 Deng)'.format(point_player))
                    player_win = player_win + 1
                    win_stu = 'Player'
                else:
                    print('Player win Pok{}'.format(point_player))
                    player_win = player_win + 1
                    win_stu = 'Player'
            elif point_player < point_bot: 
                print('Your Point :' , point_player, '\nYour Card : ' , CardPlayer)
                print('Bot Point :' , point_bot, '\nBot Card : ' , CardBot)
                if bot[0]['Suit'] == bot[1]['Suit'] or bot[0]['Card'] == bot[1]['Card']:
                    print('Bot Win Pok{} (2 Deng)'.format(point_bot))
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
                else:
                    print('Bot Win Pok{}'.format(point_bot))
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
            elif point_player == point_bot:
                print('Your Point :' , point_player, '\nYour Card : ' , CardPlayer)
                print('Bot Point :' , point_bot, '\nBot Card : ' , CardBot)
                print('Draw (Pok)')
        else:
            player_third_card = 1
    if x == 3:
        print('Your Point :' , point_player, '\nYour Card : ' , CardPlayer)
        print('Bot Point :' , point_bot, '\nBot Card : ' , CardBot)

        psort = []
        for i in player:
            psort.append(i['Card'])
        psort.sort()
        scp = psort[0]+psort[1]+psort[2]
        bsort = []
        for x in bot:
            bsort.append(x['Card'])
        bsort.sort()
        scb = bsort[0]+bsort[1]+bsort[2]

        xcard = ['23A', '234', '345', '456', '567', '678', '789', '1089', '109J', '10JQ', 'JKQ']
        xcard2 = ['JQQ', 'JKK', 'JJQ', 'JJK', 'KKQ', 'KQQ']
        tongcard = ['AAA', '222', '333', '444', '555', '666', '777', '888', '999', '101010', 'JJJ', 'QQQ', 'KKK']

        if scp in tongcard or scb in tongcard:
            if scp != scb:
                if scp in tongcard and scb not in tongcard:
                    print('Player win (Tong)')
                    player_win = player_win + 1
                    win_stu = 'Player'
                elif scb in tongcard and scp not in tongcard:
                    print('Bot win (Tong)')
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
                else:
                    print('1Draw (Tong)')
            elif scp == scb:
                print('2Draw (Tong)')
        elif scp in xcard or scb in xcard:
            if scp != scb:
                if scp in xcard and scb not in xcard:
                    if player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
                        print('Player win (Straight Flush)')
                        player_win = player_win + 1
                        win_stu = 'Player'
                    else:
                        print('Player win (Straight)')
                        player_win = player_win + 1
                        win_stu = 'Player'
                elif scb in xcard and scp not in xcard:
                    if bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
                        print('Bot win (Straight Flush)')
                        bot_win = bot_win + 1
                        win_stu = 'Bot'
                    else:
                        print('Bot win (Straight)')
                        bot_win = bot_win + 1
                        win_stu = 'Bot'
                else:
                    print('1Draw (Straight)')
            elif scp == scb:
                print('2Draw (Straight)')
        elif scp in xcard2 or scb in xcard2:
            if scp != scb:
                if scp in xcard and scb not in xcard2:
                    print('Player win (Sam lueang)')
                    player_win = player_win + 1
                    win_stu = 'Player'
                elif scb in xcard and scp not in xcard2:
                    print('Bot win (Sam lueang)')
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
                else:
                    print('1Draw (Sam lueang)')
            elif scp == scb:
                print('2Draw (Sam lueang)')
        else:
            if point_player > point_bot:
                if player[0]['Suit'] == player[1]['Suit'] and player[2]['Suit'] == '' or player[0]['Card'] == player[1]['Card'] and player[2]['Card'] == '':
                    print('Player win {} (2 Deng)'.format(point_player))
                    player_win = player_win + 1
                    win_stu = 'Player'
                elif player[0]['Suit'] == player[1]['Suit'] == player[2]['Suit']:
                    print('Player win {} (3 Deng)'.format(point_player))
                    player_win = player_win + 1
                    win_stu = 'Player'
                else:
                    print('Player win {}'.format(point_player))
                    player_win = player_win + 1
                    win_stu = 'Player'
            elif point_player < point_bot: 
                if bot[0]['Suit'] == bot[1]['Suit'] and bot[2]['Suit'] == '' or bot[0]['Card'] == bot[1]['Card'] and bot[2]['Card'] == '':
                    print('Bot win {} (2 Deng)'.format(point_bot))
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
                elif bot[0]['Suit'] == bot[1]['Suit'] == bot[2]['Suit']:
                    print('Bot win {} (3 Deng)'.format(point_bot))
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
                else:
                    print('Bot win {}'.format(point_bot))
                    bot_win = bot_win + 1
                    win_stu = 'Bot'
            elif point_player == point_bot:
                print('Draw (Normal)')
    # print(win_stu)
    return player_win, bot_win, player_third_card , win_stu